src/ctap2/commands/bio_enrollment.rs

```diff
@@ -0,0 +1,540 @@
+use crate::{
+    crypto::{PinUvAuthParam, PinUvAuthToken},
+    ctap2::server::UserVerificationRequirement,
+    errors::AuthenticatorError,
+    transport::errors::HIDError,
+    FidoDevice, Pin,
+};
+use serde::{
+    de::{Error as SerdeError, IgnoredAny, MapAccess, Visitor},
+    ser::SerializeMap,
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use serde_bytes::ByteBuf;
+use serde_cbor::{from_slice, to_vec, Value};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+use std::fmt;
+
+use super::{Command, CommandError, PinUvAuthCommand, RequestCtap2, StatusCode};
+
+#[derive(Debug, Clone, Copy, Serialize_repr, Deserialize_repr)]
+#[repr(u8)]
+pub enum BioEnrollmentModality {
+    Fingerprint = 0x01,
+}
+
+pub type BioTemplateId = Vec<u8>;
+#[derive(Debug, Clone, Deserialize, Default)]
+struct BioEnrollmentParams {
+    template_id: Option<ByteBuf>,           // Template Identifier.
```

Comment:
Hmm... We currently hand out `ByteBuf` in the `BioEnrollmentResponse`, so I also use that for the input (as users will not create their own ID, but use one returned by a previous Enrollment-Request).
But I agree that this is somewhat ugly. I think it would be best to introduce a new `TemplateId`-type for this (which is a wrapper around `Vec<u8>`).